packages/venia-ui/lib/RootComponents/Category/category.js

```diff
@@ -18,6 +18,16 @@ const Category = props => {
     const { currentPage, totalPages } = paginationValues;
     const { setCurrentPage, setTotalPages } = paginationApi;
 
+    const [sort, setSort] = useState({
+        sortAttribute: 'relevance',
+        sortDirection: 'ASC'
+    });
```

Comment:
Generally, we avoid storing objects in state unless they're stored outside of the component as constants.

```jsx
const Foo = props => {
  const [bar, setBar] = useState({ a: 1, b: 2 })

  // causes an infinite loop
  useEffect(() => {
    // what if the next state is the "same" as the current one?
    const nextBar = { a: 1, b: 2 }

    // always true, since they're different objects
    if (nextBar !== bar) {
      setBar(nextBar)
    }
  }, [bar, setBar])

  return <div>{`${bar.a} ${bar.b}`}<div>
}
```

We want to be able to know when the state has changed and when it hasn't, so it's better to use a boolean, number, or string if possible. Coincidentally, if we have a collection of objects, we generally need to derive a key for each object, and that key should also be a boolean, number, or string. So deriving a proper key for each object can also give you a useful value to store in state.

Reply:
Main Problem we should have default `sortState`  on every category page and i dont know solve this better.

Reply:
We need `attribute` and  `direction` to build a valid graph query 

```
 sortAttribute: 'relevance',
 sortDirection: 'ASC'
```

Reply:
> We need attribute and direction to build a valid graph query

Yes, right. I was merely suggesting that we store the state differently. Honestly though, it's not a big deal. This is probably fine.